File: c++/20291.cpp
#include <iostream>
#include <map>
#include <string>

using namespace std;

map<string, int> dic;

int main() {
  freopen("example.txt", "r", stdin);

  int N;
  cin >> N;

  string s;

  while (N--) {
    cin >> s;
    string ext = s.substr(s.find('.') + 1);
    if (dic[ext] == 0) {
      dic[ext] = 1;
    } else {
      dic[ext]++;
    }
  }

  for (auto iter = dic.begin(); iter != dic.end(); iter++) {
    cout << iter->first << ' ' << iter->second << '\n';
  }

  return 0;
}

File: js/20291.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');

const N = +inputs[0];
const dic = new Map();

for (let i = 1; i <= N; i++) {
  const [_, ext] = inputs[i].split('.');

  if (!dic.has(ext)) {
    dic.set(ext, 1);
  } else {
    dic.set(ext, dic.get(ext) + 1);
  }
}

const ans = [...dic.entries()]
  .sort()
  .map((e) => e.join(' '))
  .join('\n');

console.log(ans);
